pkg/asset/manifests/aws/zones.go

```diff
@@ -0,0 +1,272 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	capa "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/installconfig/aws"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	utilscidr "github.com/openshift/installer/pkg/asset/manifests/capiutils/cidr"
+	"github.com/openshift/installer/pkg/types"
+)
+
+type subnetsInput struct {
+	vpc            string
+	privateSubnets aws.Subnets
+	publicSubnets  aws.Subnets
+}
+
+type zonesInput struct {
+	InstallConfig *installconfig.InstallConfig
+	Cluster       *capa.AWSCluster
+	ClusterID     *installconfig.ClusterID
+	ZonesInRegion []string
+	Subnets       *subnetsInput
+}
+
+// GatherZonesFromMetadata retrieves zones from AWS API to be used
+// when building the subnets to CAPA.
+func (zin *zonesInput) GatherZonesFromMetadata() (err error) {
+	if zin.InstallConfig == nil {
+		return fmt.Errorf("failed to get installConfig: %w", err)
+	}
+	if zin.InstallConfig.AWS == nil {
+		return fmt.Errorf("failed to get AWS metadata: %w", err)
+	}
+	zin.ZonesInRegion, err = zin.InstallConfig.AWS.AvailabilityZones(context.TODO())
+	if err != nil {
+		return fmt.Errorf("failed to get availability zones: %w", err)
+	}
+	return nil
+}
+
+// GatherSubnetsFromMetadata retrieves subnets from AWS API to be used
+// when building the subnets to CAPA.
+func (zin *zonesInput) GatherSubnetsFromMetadata() (err error) {
+	if zin.InstallConfig == nil {
+		return fmt.Errorf("failed to get installConfig: %w", err)
+	}
+	if zin.InstallConfig.AWS == nil {
+		return fmt.Errorf("failed to get AWS metadata: %w", err)
+	}
+	zin.Subnets = &subnetsInput{}
+	if zin.Subnets.privateSubnets, err = zin.InstallConfig.AWS.PrivateSubnets(context.TODO()); err != nil {
+		return fmt.Errorf("failed to get private subnets: %w", err)
+	}
+	if zin.Subnets.publicSubnets, err = zin.InstallConfig.AWS.PublicSubnets(context.TODO()); err != nil {
+		return fmt.Errorf("failed to get public subnets: %w", err)
+	}
+	if zin.Subnets.vpc, err = zin.InstallConfig.AWS.VPC(context.TODO()); err != nil {
+		return fmt.Errorf("failed to get VPC: %w", err)
+	}
+	return nil
+}
+
+type zonesCAPI struct {
+	AvailabilityZones sets.Set[string]
```

Comment:
Unless I'm misunderstanding here, this is just a union of `controlPlaneZones + computeZones`? I think you can use a function instead:
```golang
func (zo *zonesCAPI) AvailabilityZones() sets.Set[string] {
    return zo.controlPlaneZones.Union(zo.computeZones)
}